mirdata/groove_midi.py

```diff
@@ -0,0 +1,320 @@
+        style (str): Style (genre, groove type) of the track (ex. 'funk/groove1') 
+        bpm (int): Track bpm (ex. 138) 
+        beat_type (str): Whether the track is a beat or a fill (ex. 'beat') 
+        time_signature (str): Time signature of the track (ex. '4-4', '6-8')  
+        midi_path (str): Path to the midi file 
+        audio_path (str): Path to the audio file 
+        duration (float): Duration of the midi file in seconds 
+        split (str): Whether the track is for a train/valid/test set. One of 'train', 'valid' or 'test'.
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError(
+                '{} is not a valid track ID in Groove MIDI'.format(track_id)
+            )
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+
+        metadata = DATA.metadata(data_home)
+        if metadata is not None and track_id in metadata:
+            self._track_metadata = metadata[track_id]
+        else:
+            self._track_metadata = {
+                "drummer": None,
+                "session": None,
+                "style": None,
+                "bpm": None,
+                "beat_type": None,
+                "time_signature": None,
+                "midi_filename": None,
+                "audio_filename": None,
+                "duration": None,
+                "split": None,
+            }
+
+        self.drummer = self._track_metadata["drummer"]
+        self.session = self._track_metadata["session"]
+        self.style = self._track_metadata["style"]
+        self.bpm = self._track_metadata["bpm"]
+        self.beat_type = self._track_metadata["beat_type"]
+        self.time_signature = self._track_metadata["time_signature"]
+        self.duration = self._track_metadata["duration"]
+        self.split = self._track_metadata["split"]
+        self.midi_filename = self._track_metadata["midi_filename"]
+        self.audio_filename = self._track_metadata["audio_filename"]
+
+        self.midi_path = os.path.join(self._data_home, self._track_paths["midi"][0])
+
+        if self._track_paths["audio"][0]:
+            self.audio_path = os.path.join(
+                self._data_home, self._track_paths["audio"][0]
+            )
+        else:
+            self.audio_path = None
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        if self.audio_path:
+            return load_audio(self.audio_path)
+        else:
+            return (None, None)
+
+    @property
+    def midi(self):
+        """(obj): prettyMIDI obj"""
+        return load_midi(self.midi_path)
+
+    def to_jams(self):
+        """(Not Implemented) Jams: the track's data in jams format"""
```

Comment:
would you like me to implement the `to_jams` method for you?

Reply:
I assumed it was more for an annotation heavy dataset. That would be great!

Reply:
Done in https://github.com/kyungyunlee/mirdata/pull/1